File: src/main/java/kr/ygh/maple/domain/union/UnionRaider.java
package kr.ygh.maple.domain.union;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import kr.ygh.maple.domain.union.subset.UnionRaiderBlock;
import kr.ygh.maple.domain.union.subset.UnionRaiderStat;

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public record UnionRaider(String date, String[] union_raider_stat, String[] union_occupied_stat,
                          UnionRaiderStat[] union_inner_stat, UnionRaiderBlock[] union_block
) {
}

File: frontend/src/domain/union/unionRaider.ts
export interface UnionRaider {
  date: string;
  union_raider_stat: string[];
  union_occupied_stat: string[];
  union_inner_stat: UnionRaiderStat[];
  union_block: UnionRaiderBlock[];
}

export interface UnionRaiderStat {
  stat_field_id: string;
  stat_field_effect: string;
}

export interface UnionRaiderBlock {
  block_type: string;
  block_class: string;
  block_level: string;
  block_control_point: UnionRaiderBlockPoint;
  block_position: UnionRaiderBlockPoint[];
}

export interface UnionRaiderBlockPoint {
  x: number;
  y: number;
}
